stats/opencensus/e2e_test.go

```diff
@@ -935,6 +937,376 @@ func (s) TestOpenCensusTags(t *testing.T) {
 			t.Fatalf("Should have received a nil error from channel, instead received: %v", chErr)
 		}
 	}
-
 	wg.Wait()
 }
+
+// compareSpanContext only checks the equality of the trace options, which
+// represent whether the span should be sampled. The other fields are checked
+// for presence in later assertions.
+func compareSpanContext(sc trace.SpanContext, sc2 trace.SpanContext) bool {
+	return sc.TraceOptions.IsSampled() == sc2.TraceOptions.IsSampled()
+}
+
+func compareMessageEvents(me []trace.MessageEvent, me2 []trace.MessageEvent) bool {
+	if len(me) != len(me2) {
+		return false
+	}
+	// Order matters here, message events are deterministic so no flakiness to
+	// test.
+	for i, e := range me {
+		e2 := me2[i]
+		if e.EventType != e2.EventType {
+			return false
+		}
+		if e.MessageID != e2.MessageID {
+			return false
+		}
+		if e.UncompressedByteSize != e2.UncompressedByteSize {
+			return false
+		}
+		if e.CompressedByteSize != e2.CompressedByteSize {
+			return false
+		}
+	}
+	return true
+}
+
+// compareLinks compares the type of link received compared to the wanted link.
+func compareLinks(ls []trace.Link, ls2 []trace.Link) bool {
+	if ls == nil && ls2 == nil {
+		return true
+	}
+	if ls == nil || ls2 == nil {
+		return false
+	}
+	if len(ls) != len(ls2) {
+		return false
+	}
+	for i, l := range ls {
+		l2 := ls2[i]
+		if l.Type != l2.Type {
+			return false
+		}
+	}
+	return true
+}
+
+// spanInformation is the information received about the span. This is a subset
+// of information that is important to verify that gRPC has knobs over, which
+// goes through a stable OpenCensus API with well defined behavior. This keeps
+// the robustness of assertions over time.
+type spanInformation struct {
+	// SpanContext either gets pulled off the wire in certain cases server side
+	// or created.
+	sc              trace.SpanContext
+	parentSpanID    trace.SpanID
+	spanKind        int
+	name            string
+	message         string
+	messageEvents   []trace.MessageEvent
+	status          trace.Status
+	links           []trace.Link
+	hasRemoteParent bool
+	childSpanCount  int
+}
+
+// presenceAndRelationshipAssertionsClientServerSpan checks for consistent trace
+// ID across the full trace. It also asserts each span has a corresponding
+// generated SpanID, and makes sure in the case of a server span and a client
+// span, the server span points to the client span as it's parent. This is
```

Comment:
*its

And at least a couple other places in this pr.

It's = it is
Its = possessive pronoun, just like how "his" doesn't use an apostrophe.